### PR TITLE
[WIP]Add attach-disk method for npiv virtual disk

### PR DESCRIPTION
**Pull request by lento-sun**
Previously we only use attach-device method, now add attach-disk
method according to some testcases.

Sign-offed by: Yi Sun (yisun@redhat.com)

Now my NPIV env has some problems, so cannot verify this pr, we'll udpate test run results when NPIV machine ok.

### CODE REVIEW

**Comment by lento-sun** [comment]
seems a HBA card driver issue, opened a bug to track:
https://bugzilla.redhat.com/show_bug.cgi?id=1476227

**Comment by kylazhang** [comment]
@lento-sun please update the latest status, if we can review now, please give result and remove WIP, thanks

File: libvirt/tests/cfg/npiv/npiv_virtual_disk.cfg [hunk 1]
```diff
@@ -31,3 +32,7 @@
                             device_target = "vdb"
                             target_bus = "virtio"
                             readonly = "yes"
+    variants:
+        - attach_device:
```
Comment (dzhengfy) [inline libvirt/tests/cfg/npiv/npiv_virtual_disk.cfg]: If you move line 9 after this line, I think it is more pretty.

**Comment by iccaszhulili** [comment]
rh-caselink retest please

File: libvirt/tests/cfg/npiv/npiv_virtual_disk.cfg (hunk 1 above)
Comment (lento-sun) [reply libvirt/tests/cfg/npiv/npiv_virtual_disk.cfg]: updated, thx

**Comment by lento-sun** [comment]
Close for now, @jgao12345 will take over npiv job when having machines for autocase writing.

**Pull request closed without merging**